### ISSUE TITLE
Update to some unknown properties

### ISSUE THREAD

**Issue by nygard** [amoeba/treestats.net#12]
Here are the ones I know about:

139 (0x8b)  = 0xe9 - Player Level
199 (0xc7)  = 0x44c1f8ef - House Purchase Date (seconds since 1970)

0x11f: Celestial Hand Society Standing
0x120: Eldrytch Web Society Standing
289 (0x121) = 0x1 - Radiant Blood Society Standing
    if (standing < 101)   return @"Initiate";
    if (standing < 301)   return @"Adept";
    if (standing < 601)   return @"Knight";
    if (standing < 1001)  return @"Lord";
    if (standing == 1001) return @"Master";
The value (standing) is the number of ribbons you've turned in.

307 (0x133) = 0x0 - Damage Rating
308 (0x134) = 0x0 - Damage Reduction Rating
314 (0x13a) = 0x0 - Crit Damage Rating
322 (0x142) = 0x7 - Unlocked Aetheria Slots (1 blue, 2 yellow, 4 red)
323 (0x143) = 0x0 - Healing Rating // From Growth Aetheria
350 (0x15e) = 0x0 - Damage Over Time Reduction Rating
351 (0x15f) = 0x0 - Life Reduction Rating // vigor

**Comment by amoeba** [comment]
Thanks for sending those along! I'll take a look soon and integrate them.

**Comment by amoeba** [comment]
I got a lot of my DWORD props from this pastebin, http://pastebin.com/X05rYnYU. There are some big differences between yours and the above paste. So far, yours looks like it has some plausible stuff in it.

**139 (0x8b) = 0xe9 - Player Level**
Looks right, thanks!

**199 (0xc7) = 0x44c1f8ef - House Purchase Date (seconds since 1970**
Looks right, thanks!

**0x11f: Celestial Hand Society Standing**
Decimal 287, I have this down as "Ribbons", assuming they meant ribbon count? My value of 287=>Ribbons looks legit but I don't know the game well enough anymore. (http://treestats.net/Morningthaw/Mr%20Adventure)[Mr Adventure-Morningthaw] has a Society value of 1 and a Ribbons values of 1001. That 1001 seems like the right value for Ribbons counts because I saw somewhere that 1001 is the maximum value for ribbons and it's a strange number for it to be anything else.

**0x120: Eldrytch Web Society Standing**
Decimal 288, I don't have this one

**289 (0x121) = 0x1 - Radiant Blood Society Standing**
This one makes sense. My character, [Kolthar al Magus-Wintersebb](http://treestats.net/WintersEbb/Kolthar%20al%20Magus), has DWORD 289 set to 1 and I think he's a Radiant Blood member and not high at all.

As for these,

```
307 (0x133) = 0x0 - Damage Rating
308 (0x134) = 0x0 - Damage Reduction Rating
314 (0x13a) = 0x0 - Crit Damage Rating
322 (0x142) = 0x7 - Unlocked Aetheria Slots (1 blue, 2 yellow, 4 red)
323 (0x143) = 0x0 - Healing Rating // From Growth Aetheria
350 (0x15e) = 0x0 - Damage Over Time Reduction Rating
351 (0x15f) = 0x0 - Life Reduction Rating // vigor
```

I have totally different number mappings, 370-379. Did you VTPropertyDump these? Characters that've been uploaded that likely have ratings don't have these DWORDs so that's confusing. My DWORD mappings are in this helper, https://github.com/amoeba/treestats.net/blob/master/helpers/properties_helper.rb

Thanks for taking the time to send me these! I'll leave this open so we can chat about the differences we have for these mappings.

**Comment by nygard** [comment]
Yes, the standings, as I call them, are the number of ribbons you've turned in to a particular society.  As you reach certain thresholds, your rank increases (Initiate, Adept, Knight, Lord, Master).  I have characters in each society so I could verify this.

It's been quite a while since I've looked at these things, but I logged back in to do some testing.  The lower range of properties that I've shown for the ratings are indeed what you'll see on characters.  The higher range of properties (370 to 379) are what you'll see on items.  Summoning was added after character ratings, and the essences used for summoning may have ratings, which use that higher range.  The top end armor can also have ratings.

I've been using my own tools to look at the protocol messages, but vtpropertydump should show the same values.  When a character doesn't have a particular rating, the properties are just absent.

BTW, I expect that property 0x186 (390) is the number of times a character has been enlightened.

I hope this helps!

**Comment by amoeba** [comment]
Thanks so much! It was hard to develop parts of this without being current on how the game works. 

Why do you think a player like http://treestats.net/Morningthaw/Mr%20Adventure has non-zero values for the 370-379 DWORDs? If they're item ratings then shouldn't they be zero for a player?

Looking at one of my chars, I can confirm that 
    322 (0x142) = 0x7 - Unlocked Aetheria Slots (1 blue, 2 yellow, 4 red)
makes sense. 

Do you have any characters with non-zero values for the ratings that you could upload?

**Comment by nygard** [comment]
I'm not sure why you're seeing properties 370-379 on players -- based on my limited testing last night, I didn't have those properties on myself.

I'd upload some of my characters, but I can't get your plugin to work.  When I try to add the TreeStats.dll to Decal, I simply get "DenAgent: Failed to register plugin.", which isn't very helpful.  My decal export is at https://gist.github.com/nygard/0d92e2e7467607b56f2c in case you can see anything that might be missing.

**Comment by amoeba** [comment]
```
I'm not sure why you're seeing properties 370-379 on players -- based
```

on my limited testing last night, I didn't have those properties on myself.

That is really weird then.

```
I can't get your plugin to work
```

Hrm! Do you run other plugins successfully? I'll ask in #acdev w/ your dump.

**Comment by nygard** [comment]
Ok, I got TreeStats to work.  I wasn't using the latest version of decal before.

I've uploaded one of my characters, Lothnar (Leafcull).  He currently has
Dmg/CritDmg Rating: 11/3
Dmg/CritDmg Resist: 19/0
And here's the message the server sends when I identify myself:
https://gist.github.com/nygard/e6ed2f03db7feb4b917c

And it looks like you have dupes for properties 243-246 in helpers/properties_helper.rb -- the first set is correct.  (Lothnar augmented Fire and Lightning, website shows Fire/Piercing because of the dupes.)

**Comment by amoeba** [comment]
You are kicking ass. Thanks so much for doing all this work.  I'll integrate this tonight.
